### PR TITLE
[cxx] Fixes for all lanes to build as C++ but w/o configure.ac change.

### PR DESCRIPTION
**Pull request by jaykrell**
This is https://github.com/mono/mono/pull/17325 but with configure.ac change removed.
Each of the commits can also be separated and have been.

### CODE REVIEW

**Comment by jaykrell** [comment]
@monojenkins build failed

**Comment by jaykrell** [comment]
@monojenkins build failed

**Comment by jaykrell** [comment]
@monojenkins build failed

**Comment by jaykrell** [comment]
@monojenkins build failed

**Comment by jaykrell** [comment]
@monojenkins build failed

**Comment by jaykrell** [comment]
@monojenkins build failed

File: mono/mini/mini.h [hunk 1]
```diff
@@ -783,16 +783,34 @@ struct MonoCallInst {
...
+	CallInfo *call_info;
+#ifdef ENABLE_LLVM
+	LLVMCallInfo *cinfo;
+	int rgctx_arg_reg, imt_arg_reg;
+#endif
+#ifdef TARGET_ARM
+	/* See the comment in mini-arm.c!mono_arch_emit_call for RegTypeFP. */
+	GSList *float_args;
+#endif
+	// Bitfields are at the end due to an unexplained problem with C++ and LLVM.
```
Comment (jaykrell) [inline mono/mini/mini.h]: It is explained and fixed actually.

Comment (jaykrell) [reply mono/mini/mini.h]: 1. One of the files wasn't compiling as C++. automake typo, sort of.
2. Fixed twice: the compilation was fixed, and the ifdef around MonoJitCallId was removed, so it'll work even with that mistake.

Comment (lewurm) [reply mono/mini/mini.h]: oops.  so should we revert the commit?

Comment (akoeplinger) [reply mono/mini/mini.h]: Comment was reworked in https://github.com/mono/mono/commit/c32414966c76a10741949d4f40e0dc737753e44f.

**Pull request merged**